services/button.c

```diff
@@ -25,20 +33,19 @@ static void update(srv_t *state) {
         pin_set(state->backlight_pin, state->pressed);
         if (state->pressed) {
             jd_send_event(state, JD_BUTTON_EV_DOWN);
-            state->press_time = now;
+            state->press_time = now / 1000;
```

Comment:
the internal timestamps should be in `us`; this way overflows won't affect you (unless someone holds the button for over an hour); after subtraction of timestamps you can convert to `ms`

Reply:
got it.